test/sse.spec.js

```diff
@@ -0,0 +1,243 @@
+//modules
+var _ = require('lodash');
+var chai = require('chai');
+var ess = require('event-source-stream');
+var EventSource = require('eventsource');
+var expect = chai.expect;
+var express = require('express');
+var jsc = require('jsverify');
+var Promise = require('bluebird');
+var request = require('request');
+var sinon = require('sinon');
+var sinonChai = require('sinon-chai');
+var sse = require('tiny-sse');
+var redis = require('then-redis');
+
+chai.use(sinonChai);
+
+//locals
+var app = express();
+var Consumer = require('../lib/index.js');
+var consumer;
+var port = 3000;
+var redisUrl = 'tcp://localhost:6379';
+var redisClient = redis.createClient(redisUrl);
+var streamURL = 'http://localhost:'+port+'/stream';
+var streamThreeURL = 'http://localhost:'+port+'/streamThree';
+var streamSixURL = 'http://localhost:'+port+'/streamSix';
+
+describe('SSE Checkpointing Consumer module', function() {
+
+    before(function() {
+        createStreamRoute(app, '/stream', 1, 'bar');
+        createStreamRoute(app, '/streamThree', 3, 'bar');
+        createStreamRoute(app, '/streamSix', 6, 'bar');
+
+        var server = app.listen(port, function() {
+            var host = server.address().address;
+            var port = server.address().port;
+            console.log('Test server listening at http://%s:%s', host, port);
+        });
+    });
+
+    beforeEach(function() {
+        consumer = new Consumer(redisUrl);
```

Comment:
The consume function needs to be able to pass the last checkpoint id to the streamConstructor, which it needs to retrieve from Redis.